src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -571,33 +574,68 @@ public enum CompareExchangeTombstonesCleanupState
             NoMoreTombstones
         }
 
-        private CompareExchangeTombstonesCleanupState GetMaxCompareExchangeTombstonesEtagToDelete(TransactionOperationContext context, string databaseName,
-            DatabaseObservationState state, out long maxEtag)
+        private CompareExchangeTombstonesCleanupState GetMaxCompareExchangeTombstonesEtagToDelete(DatabaseObservationState state, out long maxEtag)
         {
             maxEtag = -1;
+            long minClusterWideTransactionIndex = -1;
+
+            // we are checking this here, not in the main loop, to avoid returning 'NoMoreTombstones' when maxEtag is 0
+            foreach (var nodeTag in state.DatabaseTopology.AllNodes)
+            {
+                if (state.Current.ContainsKey(nodeTag) == false) // we have a state change, do not remove anything
+                    return CompareExchangeTombstonesCleanupState.InvalidDatabaseObservationState;
+            }
 
-            var periodicBackupTaskIds = state.RawDatabase.PeriodicBackupsTaskIds;
-            if (periodicBackupTaskIds != null && periodicBackupTaskIds.Count > 0)
+            foreach (var nodeTag in state.DatabaseTopology.AllNodes)
             {
-                foreach (var taskId in periodicBackupTaskIds)
+                var hasState = state.Current.TryGetValue(nodeTag, out var nodeReport);
+                
+                Debug.Assert(hasState, $"Could not find state for node '{nodeTag}' for database '{state.Name}'.");
+                
+                // ReSharper disable once ConditionIsAlwaysTrueOrFalse
+                if (hasState == false)
+                    return CompareExchangeTombstonesCleanupState.InvalidDatabaseObservationState;
+                
+                var hasReport = nodeReport.Report.TryGetValue(state.Name, out var report);
+                Debug.Assert(hasReport || nodeReport.Error != null, $"Could not find report for node '{nodeTag}' for database '{state.Name}'.");
+                // ReSharper disable once ConditionIsAlwaysTrueOrFalse
+                if (hasReport == false)
+                    return CompareExchangeTombstonesCleanupState.InvalidDatabaseObservationState;
+
+                if (state.RawDatabase.PeriodicBackupsTaskIds != null && state.RawDatabase.PeriodicBackupsTaskIds.Count != 0 &&
+                    report.BackupStatuses.Count == 0)
```

Comment:
I am checking for backwards compatibilty like this instead of allowing null in BackupStatues to avoid any possible NREs